### PR TITLE
Enable cname resolution when realm is specified in connection

### PR DESCRIPTION
**Pull request by v-zhangw**
Same as https://github.com/microsoft/mssql-jdbc/pull/1688 with additional logging and running formatter.

### CODE REVIEW

**Comment by tkyc** [comment]
I'm not too familiar with the code, but wasn't there a mention of a test update required or was it an addition of a new test?

File: src/main/java/com/microsoft/sqlserver/jdbc/SSPIAuthentication.java [hunk 1]
```diff
@@ -149,23 +149,34 @@ String enrichSpnWithRealm(SQLServerConnection con, String spn, boolean allowHost
...
-                // We failed, try with canonical host name to find a better match
-                try {
+        try {
+            if (null == realm || realm.trim().isEmpty()) {
+                RealmValidator realmValidator = getRealmValidator();
+                realm = findRealmFromHostname(realmValidator, dnsName);
+                if (null == realm && allowHostnameCanonicalization) {
+                    if (logger.isLoggable(Level.FINER)) {
+                        logger.finer("Realm not provided or found from host name, attempt to find canonical host name via InetAddress function and then derive realm");
+                    }
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SSPIAuthentication.java:159]: seems a bit wordy, how about "Attempt to derive realm using canonical host name"?

File: src/main/java/com/microsoft/sqlserver/jdbc/SSPIAuthentication.java [hunk 2]
```diff
@@ -149,23 +149,34 @@ String enrichSpnWithRealm(SQLServerConnection con, String spn, boolean allowHost
...
+                    // Match means hostname is correct (eg if server name was an IP) so override dnsName as well
                     dnsName = canonicalHostName;
-                } catch (UnknownHostException e) {
-                    // ignored, cannot canonicalize
+                }
+            } else {
+                if (allowHostnameCanonicalization) {
                     if (logger.isLoggable(Level.FINER)) {
-                        logger.finer("Could not canonicalize host name. " + e.toString());
+                        logger.finer("Since realm is provided, attempt to get canonical host name with InetAddress function");
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SSPIAuthentication.java:169]: log message here seems a bit misleading, what we are trying to do as per comment below is to resolve cname to hostname

**Review by lilgreenbird — CHANGES_REQUESTED** [review]
add test

**Comment by rjeggensentry** [comment]
I'm not too familiar with the project, but I didn't find any existing test cases for Kerberos authentication or this method.  Are there any test cases that could be extended to facilitate a new test case without breaking existing test capabilities?

**Review by tkyc — APPROVED** [review]

**Review by lilgreenbird — APPROVED** [review]
